src/3.1/docker-entrypoint.sh

```diff
@@ -184,17 +153,22 @@ done
 
 # Chown the data dir now that (maybe) an initial password has been
 # set (this is a file in the data dir)
-if [[ "$(stat -c %u /data)" = "0" ]]; then
-  chown -R "${user_name}:${group_name}" /data
-fi
+chown -R "${userid}":"${groupid}" /data
+chmod -R 700 /data
 
 [ -f "${EXTENSION_SCRIPT:-}" ] && . ${EXTENSION_SCRIPT}
 
 # Use su-exec to drop privileges to neo4j user
 # Note that su-exec, despite its name, does not replicate the
 # functionality of exec, so we need to use both
-if [ "${cmd}" == "neo4j" ]; then
-    exec su-exec "${user_name}" neo4j console
+if [ "${cmd}" == "neo4j" ] && [ "$(id -u)" = "0" ]; then
+  exec su-exec neo4j neo4j console
+elif [ "${cmd}" == "neo4j" ]; then
+  # docker running with --user
+  exec neo4j console
+elif [ "$(id -u)" = "0" ]; then
+  exec su-exec neo4j "$@"
 else
-    exec su-exec "${user_name}" "$@"
+  # docker running with --user
+  exec "$@"
```

Comment:
I suggest simplifying all this by creating an `${exec_cmd}` variable in the `if [ "$(id -u)" = "0" ]` switch right at the top and then applying that unconditionally to each of these cases. Would that work?

Reply:
can try